src/Shared/runtime/Http2/Hpack/DynamicTable.cs

```diff
@@ -47,6 +55,13 @@ public DynamicTable(int maxSize)
 
         public void Insert(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
         {
+            Insert(name, value);
```

Comment:
Did you forget the null index? This looks like it's calling itself.

Reply:
Yes, you're right. There is a microbenchmark that hangs. I bet this is the reason why.

I'm surprised other tests aren't broken.